lib/subproviders/geth_api_double.js

```diff
@@ -574,6 +574,18 @@ GethApiDouble.prototype.debug_traceTransaction = function(txHash, params, callba
   this.state.queueTransactionTrace(txHash, params, callback);
 };
 
+// Unlocks any account after server start
+GethApiDouble.prototype.debug_unlockAccount = function(address, callback) {
+  // Checks if given address is already unlocked
+  if (this.state.unlocked_accounts[address.toLowerCase()] !== true) {
+    // Unlocks it
+    this.state.unlocked_accounts[address.toLowerCase()] = true;
```

Comment:
Before we can merged this the behavior needs to be changed so that it _does not_ unlock accounts found in `this.state.personal_accounts`, as these can already be unlocked (if previously locked at start up or via `personal_lockAccount`) via `personal_unlockAccount`.

Reply:
In the case where the account cannot be unlocked, return an error in the callback:

```
  callback(new Error("cannot unlock known/personal account"));
```

Reply:
Resolved in this [commit](https://github.com/trufflesuite/ganache-core/pull/622/commits/010e21e05210423bcce925c8fdd7167745a8e2d0)